src/main/groovy/org/transmartproject/batch/browsetag/BrowseTagAssociationToI2b2Writer.groovy

```diff
@@ -0,0 +1,58 @@
+package org.transmartproject.batch.browsetag
+
+import groovy.util.logging.Slf4j
+import org.springframework.batch.item.ItemWriter
+import org.springframework.batch.item.file.FlatFileHeaderCallback
+import org.springframework.batch.item.file.FlatFileItemWriter
+import org.springframework.batch.item.file.transform.DelimitedLineAggregator
+import org.springframework.batch.item.file.transform.FieldExtractor
+import org.springframework.core.io.Resource
+
+import javax.annotation.PostConstruct
+
+/**
+ *
+ */
+@Slf4j
+class BrowseTagAssociationToI2b2Writer implements ItemWriter<BrowseTagAssociation> {
+
+    @Delegate
+    FlatFileItemWriter<BrowseTagValue> delegate
+
+    class BrowseTagAssociationFieldExtractor implements FieldExtractor<BrowseTagAssociation> {
+
+        Object[] extract(BrowseTagAssociation item) {
+            ['\\',
+             item.value.type.displayName,
+             item.value.description,
+             0
+            ]
+        }
+
+    }
+
+    private final Resource resource
+
+    BrowseTagAssociationToI2b2Writer(Resource resource) {
+        this.resource = resource
+    }
+
+    @PostConstruct
+    void init() {
+        delegate = new FlatFileItemWriter(
+                resource: resource,
+                lineAggregator: new DelimitedLineAggregator<BrowseTagAssociation>(
+                        delimiter: '\t',
+                        fieldExtractor: new BrowseTagAssociationFieldExtractor(),
+                ),
+                headerCallback: new FlatFileHeaderCallback() {
+                    @Override
+                    void writeHeader(Writer writer) throws IOException {
+                        writer.write(['concept_key', 'tag_title', 'tag_description', 'index'].join('\t'))
+                    }
+                }
```

Comment:
In Groovy this is done with `fieldExtractor: { ['\\', it.value.type.displayName, it.value.description, 0] } as FieldExtractor` and `headerCallback: { it.write(['concept_key', 'tag_title', 'tag_description', 'index'].join('\t')) } as FlatFileHeaderCallback`.

Plus, there's no need to have a wrapper class just for instantiating a  `FlatFileItemWriter`. Either do this instantiation directly in the `@Configuration` class, or, if you really want to have a separate class, implement `FactoryBean`.